src/components/projects/view/application-tile.svelte

```diff
@@ -1,107 +1,149 @@
 <script>
-  import { onMount, createEventDispatcher } from 'svelte';
-  import AccordionSection from 'ui/accordion-section.svelte';
-  import Button from 'ui/button.svelte';
-  import CopyButton from '@/components/projects/view/copy-button.svelte';
+  import { onMount, createEventDispatcher, getContext } from 'svelte';
+  import { stores } from '@sapper/app';
+  import { AccordionSection, Button, Popover } from 'attractions';
+  import { PopoverPositions } from 'attractions/popover';
+  import { snackbarContextKey } from 'attractions/snackbar';
+  import CopyButton from '@/components/common/copy-button.svelte';
+  import Stars from '@/components/common/stars.svelte';
+  import ViewReportsDialog from '@/components/projects/view/view-reports-dialog.svelte';
   import ApplicationStatuses from '@/constants/backend/application-statuses.js';
   import * as api from '@/utils/api.js';
 
+  const { session } = stores();
+
   export let activity;
   export let application;
-  export let handlePanelOpen;
-  let report = null;
 
+  let report = null;
   onMount(() => {
     report = api.get(
       `/projects/${activity.project}/activities/${activity.id}`
       + `/applications/${application.id}/report_info`,
-    ).then(resp => resp.json());
+    ).then(resp => resp.json()).catch(console.error);
   });
 
+  const viewReportsDialog = {
+    open: false,
+    reports: null,
+    async show({ detail }) {
+      viewReportsDialog.reports = (await report).reports;
```

Comment:
Now this is a cool example of when the Null Object design pattern can be useful 😉, given the previous comment

Reply:
And this will not even be called if the promise fails, because the button won't be there

Reply:
Yeah, but that was relevant to the `catch` resolving thing.

Reply:
Yeah, but the comment was more relevant with the `catch` resolving thing